pkg/network/setup/netconf.go

```diff
@@ -23,26 +23,51 @@ import (
 	"fmt"
 	"sync"
 
+	"kubevirt.io/kubevirt/pkg/network/netns"
+
 	v1 "kubevirt.io/api/core/v1"
 	"kubevirt.io/kubevirt/pkg/network/cache"
 )
 
 type NetConf struct {
 	setupCompleted    sync.Map
 	ifaceCacheFactory cache.InterfaceCacheFactory
+	nsFactory         nsFactory
 }
 
-func NewNetConf(ifaceCacheFactory cache.InterfaceCacheFactory) *NetConf {
-	return &NetConf{
+type nsFactory func(int) NSExecutor
```

Comment:
I see what you're doing, but I think this is misleading: an `nsFactory` "creates" `ns`s. You're not creating namespaces. You're tapping into an existing namespace from another. At most, you have a connection to an NS, or a way to execute arbitrary code in that namespace. 

Plus, would we want to be more specific ? i.e. there are 6 namespace types - this one manages what it needs - net namespaces. Since all usages are within the pkg, I don't think this is that important.

Reply:
An `nsFacotry` creates an object that represent a namespace.

Regarding the specific name, I do not think we need to be specific.. In theory, this factory may represent any namespace type. We populate it with a specific one, which is the network one.